timesketch/lib/analyzers/ioc_extraction.py

```diff
@@ -0,0 +1,280 @@
+            'optional': True
+        }
+    ]
+
+
+    def __init__(self, index_name, sketch_id, config=None):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+            config: Optional dict that contains the configuration for the
+                analyzer. If not provided, the default YAML file will be
+                loaded up.
+        """
+        self.index_name = index_name
+        super(IOCExtractionSketchPlugin, self).__init__(
+            index_name, sketch_id)
+        self._config = config
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result.
+        """
+        config = self._config or interface.get_yaml_config(self.CONFIG_FILE)
+        if not config:
+            return 'Unable to parse the config file.'
+        query, ret_fields, ioc_db = self.parse_config(config)
+        if not query or not ret_fields:
+            return 'Error to make query'
+        if not ioc_db:
+            return 'Error to parse config'
+        logger.info((
+            'Return field '
+            'error: {0!s}').format(ret_fields))
+        return self.extract_ioc(query, ret_fields, ioc_db)
+
+    def parse_config(self, ioc_config):
+        """Create IOC DB and make request
+
+        Args:
+            config: A dict that contains the configuration for the ioc
+                extraction. See data/ioc.yaml for fields and further
+                documentation of what needs to be defined.
+
+        Returns:
+            query: A string that contains elastic query to return
+                events check ioc against.
+            return_fields: A list that contains names of field to return
+                from elastic request to check ioc against.
+            ioc_db: A dict contains extracted config and IOC files
+        """
+        ioc_db = []
+        query = ""
+        return_fields = []
+        list_fieldnames = self.get_fields_list()
+        for name, config in iter(ioc_config.items()):
+            elm = {'name': name, 'found': 0}
+            path_file_ioc = config.get('path_file_ioc')
+            if not path_file_ioc:
+                logger.warning('No path for file IOC defined.')
+                continue
+            if not os.path.isfile(path_file_ioc):
+                logger.warning('file IOC defined not exists.')
+                return ''
+            elm['ioc'] = None
+            with open(path_file_ioc) as json_file:
+                try:
+                    elm['ioc'] = json.load(json_file)
+                    if not isinstance(elm['ioc'], (list, tuple)):
+                        continue
+                    if not elm['ioc']:
+                        continue
+                except json.JSONDecodeError as exception:
+                    logger.warning((
+                        'File IOC error to parse: '
```

Comment:
here you can first of all include what IOC the error applies to (helps with debugging) also you can include `exc_info=True` so that the stacktrace is included